public/app/core/services/echo/Echo.ts

```diff
@@ -66,8 +67,16 @@ export class Echo implements EchoSrv {
     });
   };
 
+  setMetaExtensions = (extensions: Record<string, unknown>) => {
+    this._metaExtensions = extensions;
+    return () => {
+      this._metaExtensions = {};
+    };
```

Comment:
If we set this API, it means if two independent objects call this callback it will unexpectedly remove the other component metadata.

Why don't we just delegate the responsibility of removing the meta extensions to the service consumer?

Reply:
> If we set this API, it means if two independent objects call this callback it will unexpectedly remove the other component metadata.

Yes, this is true. But I think of this API as a low-level, Echo config. Can add some warnings and more verbose docs on how this API behaves. But once meta extensions are set, no new extensions should be possible to add.

For one-off meta modifications we have an API already in the `reportInteraction` itself.

> Why don't we just delegate the responsibility of removing the meta extensions to the service consumer?

What do you mean?

Reply:
> Can add some warnings and more verbose docs on how this API behaves. 

➕ 1 to this idea :)

Reply:
@ivanortegaalba  - I've opened a simpler, less invasive alternative: https://github.com/grafana/grafana/pull/79963